x/ecocredit/basket/keeper/query_fees.go

```diff
@@ -22,7 +22,7 @@ func (k Keeper) BasketFee(ctx context.Context, req *types.QueryBasketFeeRequest)
 		var ok bool
 		fee, ok = regentypes.ProtoCoinToCoin(basketFee.Fee)
 		if !ok {
-			return nil, sdkerrors.ErrInvalidType.Wrapf("failed to parse basket fee")
+			return nil, regenerrors.ErrInvalidArgument.Wrap("failed to parse basket fee")
```

Comment:
Should this be internal? The user is not providing an invalid argument.

```suggestion
			return nil, regenerrors.ErrInternal.Wrap("failed to parse basket fee")
```